sector-base/src/io/fr32.rs

```diff
@@ -635,20 +635,25 @@ pub fn clear_right_bits(byte: &mut u8, offset: usize) {
     *(byte) &= !((1 << offset) - 1)
 }
 
-pub fn write_padded<W: ?Sized>(source: &[u8], target: &mut W) -> io::Result<usize>
+const N: usize = 1000;
+const CHUNK_SIZE: usize = 127 * N;
+
+pub fn write_padded<R, W>(source: &mut R, target: &mut W) -> io::Result<usize>
 where
-    W: Read + Write + Seek,
+    R: Read + ?Sized,
+    W: Read + Write + Seek + ?Sized,
 {
     // In order to optimize alignment in the common case of writing from an aligned start,
     // we should make the chunk a multiple of 127 (4 full elements, see `PaddingMap#alignment`).
-    // n was hand-tuned to do reasonably well in the benchmarks.
-    let n = 1000;
-    let chunk_size = 127 * n;
-
+    // N was hand-tuned to do reasonably well in the benchmarks.
+    let mut buffer = [0; CHUNK_SIZE];
     let mut written = 0;
 
-    for chunk in source.chunks(chunk_size) {
-        written += write_padded_aux(&FR32_PADDING_MAP, chunk, target)?;
+    while let Ok(bytes_read) = source.read(&mut buffer) {
+        if bytes_read == 0 {
+            break;
+        }
+        written += write_padded_aux(&FR32_PADDING_MAP, &buffer[..bytes_read], target)?;
```

Comment:
(another one just for me) the problem with the previous implementation was that we were allocating a new (temporary) `CHUNK_SIZE` buffer in every `chunks` call of the loop, right?